lib/collection/src/shards/queue_proxy_shard.rs

```diff
@@ -327,13 +388,13 @@ impl Inner {
     /// idempotent.
     async fn transfer_wal_batch(&self) -> CollectionResult<bool> {
         let mut update_lock = Some(self.update_lock.lock().await);
-        let start_index = self.last_update_idx.load(Ordering::Relaxed) + 1;
+        let transfer_from = self.transfer_from.load(Ordering::Relaxed);
 
         // Lock wall, count pending items to transfer, grab batch
         let (pending_count, batch) = {
             let wal = self.wrapped_shard.wal.lock();
-            let items_left = wal.last_index().saturating_sub(start_index - 1);
-            let batch = wal.read(start_index).take(BATCH_SIZE).collect::<Vec<_>>();
+            let items_left = wal.last_index().saturating_sub(transfer_from - 1);
```

Comment:
1) Do we still need to do `transfer_from - 1`, if we don't do `start_index = self.last_update_idx + 1` anymore?
2) If we do still need `-1`, then what if `transfer_from = 0`?

Reply:
I think it should be like this now:

```rust
let items_left = (wal.last_index() + 1).saturating_sub(transfer_from);
```

So if:
- `transfer_from` is 8
- `wal.last_index` is 12

We have `(12 + 1) - 8 = 5`. Which matches transferring 8, 9, 10, 11, 12.

What do you think?

Reply:
Yeah, seems reasonable. 👌